include/core/specific_heat_model.h

```diff
@@ -114,11 +144,85 @@ class PhaseChangeSpecificHeat : public SpecificHeatModel
       }
   }
 
+
+  /**
+   * @brief vector_value Calculates the vector of specific heat.
+   * @param field_vectors Vector of fields on which the specific heat may depend.
+   * @param property_vector Vector of specific_heat values.
+   */
+  virtual void
+  vector_value(const std::map<field, std::vector<double>> &field_vectors,
+               std::vector<double> &property_vector) override
+  {
+    const std::vector<double> &temperature_vec =
+      field_vectors.at(field::temperature);
+    const std::vector<double> &previous_temperature_vec =
+      field_vectors.at(field::previous_temperature);
+
+    const unsigned int n_values = temperature_vec.size();
+
+    Assert(n_values == previous_temperature_vec.size(),
+           SizeOfFields(n_values, previous_temperature_vec.size()));
+    Assert(n_values == property_vector.size(),
+           SizeOfFields(n_values, property_vector.size()));
+
+    for (unsigned int i = 0; i < n_values; ++i)
+      {
+        const double temperature          = temperature_vec[i];
+        const double previous_temperature = previous_temperature_vec[i];
+        if (temperature > previous_temperature)
```

Comment:
Isn't it better to use std::abs instead of this if and else?

Reply:
Generally it would, but in this case there are some exceptions that need to be taken into account that is going to be more robust this way. We can revisit this later though :).